test/boost/cql_query_test.cc

```diff
@@ -5348,6 +5348,30 @@ SEASTAR_TEST_CASE(test_parallelized_select_counter_type) {
     });
 }
 
+SEASTAR_TEST_CASE(test_single_partition_aggregation_is_not_parallelized) {
+    // It's pointless from performande pov to parallelize 
+    // aggregation queries which reads only single partition.
+    
+    return with_parallelized_aggregation_enabled_thread([](cql_test_env& e) {
+        auto& qp = e.local_qp();
+        auto stat_parallelized = qp.get_cql_stats().select_parallelized;
+
+        e.execute_cql("CREATE TABLE tbl (pk int, ck int, col int, PRIMARY KEY (pk, ck));").get();
```

Comment:
As I mentioned above, please expand this test to check many more cases:
1. Table with several partition key components and query that restricts them all (shouldn't be parallelized) or just one (should be parallelized)
2. Query which has **in addition** to the partition key restriction also a clustering key restriction (e.g., "ck>7"), let's check it doesn't change the parallelization decision.
3. Query which selects a single token (`token(pk)=3`) instead of a single partition. Also doesn't need parallelization.

Reply:
Added all of the cases.

> or just one (should be parallelized)

In theory this request can be parallelized but it requires `ALLOW FILTERING` clause and currently we don't allow to distribute queries via forward_service which needs filtering.

Reply:
If we think of a type of query which could be reasonably parallelized - but we don't because of an oversight or that it was just too much work and not important enough - let's consider creating an issue reminding us of that fact.